### DISCUSSION TITLE
tutorials/client-side-graphql-react/04-apollo-client

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#177] [course feedback]
# tutorials/client-side-graphql-react/04-apollo-client

https://www.apollographql.com/tutorials/client-side-graphql-react/04-apollo-client

**Comment by Zeghra** [comment]
Hi,
Great tutorial, easy to follow! Thank you for helping us understand Apollo!
My question is that which uri is correct? client uri: 'https://odyssey-lift-off-server.herokuapp.com/' or uri: "https://graphql.org/swapi-graphql", first was not accepted in Code challenge only the later one (without '/' slash at the end).

**Reply by mabuyo** [reply]
Hey there @Zeghra - For the code challenge, we are using `https://graphql.org/swapi-graphql`, as the challenge text specifies. Hope that helps!

**Comment by DreamerAI** [comment]
Hi,
I think the options for ApolloClient updated, and now there are now uri option, but this one fixed
```typescript
const httpLink = new HttpLink({
  uri: 'https://odyssey-lift-off-server.herokuapp.com/',
});

const client = new ApolloClient({
  link : ApolloLink.from([httpLink]),
  cache: new InMemoryCache(),
});
```

**Reply by DreamerAI** [reply]
Also, it seems, in official Apollo docs "uri" option still exist
